Shopify/shopify-cli
# Support running javy on arm

### WHY are these changes introduced?

Closes https://github.com/Shopify/script-service/issues/4078

### WHAT is this pull request doing?

- Detect when a mac machine is on arm and download the appropriate Javy binary
- Fix broken Javy tests that don't stub out a platform properly

### 🎩 

![image](https://user-images.githubusercontent.com/28009669/145615287-45fe3f11-2f4e-4bd3-ad7f-27b21675cfba.png)

### How to test your changes?

- On your M1 Mac, create a new typescript script. 
  - Note: You have to ensure you ran `bundle install`. For some reason before I did this, my ruby env thought it was running in `x86`. I have no idea why, but maybe it has something to do with rosetta? 🤷‍♂️ 
- Run `rm shopify-cli/ext/javy/bin/javy-v0.1.0` to remove any old installation if you have one
- Run `npm run prebuild` on your script to compile to JS
- Run `shopify-cli/bin/shopify script javy -i build/index.js -o build/index.wasm`

### Update checklist
